lib/optimize.js

```diff
@@ -106,6 +106,28 @@ class Optimizer {
         optimization_info.limit = limit;
         return true;
     }
+    
+    static optimize_sort(read, sort, graph, optimization_info) {
+        if (optimization_info && optimization_info.type) {
+            logger.debug('optimization aborting -- cannot append sort optimization to prior', optimization_info.type, 'optimization');
+            return false;
+        }
+        
+        let groupby = graph.node_get_option(sort, 'groupby');
+        if (groupby) {
```

Comment:
It's a little more canonical to say `if (graph.node_has_option(sort, 'groupby')) { ` in cases like this.